device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -101,6 +101,10 @@
     private final ProxySettings proxySettings;
     private SSLContext sslContext;
 
+    // Used to store the CorrelationCallbackMessage for a correlationId
+    private final static Map<String, CorrelatingMessageCallback> correlationCallbacks = new ConcurrentHashMap<>();
```

Comment:
These two maps shouldn't be static. Each instance of this class should have its own set of callbacks and contexts

Reply:
Got it, so there will not be a chance that the messages will be sent on one transport and received on another?